vault/resource_gcp_auth_backend.go

```diff
@@ -73,6 +73,14 @@ func gcpAuthBackendResource() *schema.Resource {
 				Optional:    true,
 				Description: "Specifies if the auth method is local only",
 			},
+			"custom_endpoint": {
```

Comment:
I wonder if we should make this a resource block rather than a map, since it seems to have defined structure.

https://github.com/hashicorp/terraform-provider-vault/blob/24044c48b6aed5d76da9d8856aa348a13a727c9b/vault/resource_rabbitmq_secret_backend_role.go#L44-L49 matches pretty closely. You will want a `MaxItems: 1` however.

Reply:
Oh nice. That looks better. I'll give it a shot 👍

Reply:
Okay, I switched things over and re-tested in https://github.com/hashicorp/terraform-provider-vault/pull/1482/commits/6f33a6f5fcbcfaa79c238d9a9d8b33f0af040f82. Let me know your thoughts.